student/src/main/java/tds/student/web/handlers/TestShellHandler.java

```diff
@@ -231,31 +236,7 @@ private String parseOutLatencies (Map<String, String> formParams) {
 
     TestOpportunity testOpp = StudentContext.getTestOpportunity ();
 
-    // get responses
-    TestManager testManager = new TestManager (testOpp);
-    testManager.LoadResponses (true);
-
-    // check if test length is met
-    testManager.CheckIfTestComplete ();
-
-    if (!testManager.IsTestLengthMet ()) {
-      // TODO mpatel - Check to see status and message in response and make sure
-      // following works
-      String message = "Review Test: Cannot end test because test length is not met.";
-      _tdsLogger.applicationError (message, "reviewTest", request, null);
-      HttpContext.getCurrentContext ().getResponse ().sendError (HttpStatus.SC_FORBIDDEN, "Cannot end the test.");
-      return null;
-    }
-
-    // check if all visible pages are completed
-    PageList pageList = testManager.GetVisiblePages ();
-
-    if (!pageList.isAllCompleted ()) {
-      String message = "Review Test: Cannot end test because all the groups have not been answered.";
-      _tdsLogger.applicationError (message, "reviewTest", request, null);
-      HttpContext.getCurrentContext ().getResponse ().sendError (HttpStatus.SC_FORBIDDEN, "Cannot end the test.");
-      return null;
-    }
+    ResponseData<String> responseData = reviewTestService.reviewTest(testOpp, request);
```

Comment:
I think we talked about this today, but the validation you're referring to is in the `ReviewTestServiceimpl`: `PageList#getVisiblePages()` is called and then `pageList.isAllCompleted()` is evaluated.